social_core/backends/base.py

```diff
@@ -116,6 +117,8 @@ def extra_data(self, user, uid, response, details=None, *args, **kwargs):
             # store the last time authentication toke place
             'auth_time': int(time.time())
         }
+        if self.GET_ALL_EXTRA_DATA or self.setting('GET_ALL_EXTRA_DATA', False):
+            self.EXTRA_DATA = [data_key for data_key in response]
```

Comment:
I literally woke up this morning thinking I should add this. Nice timing! I have a slightly alternative way of doing the same thing, to prevent us having to change too much code in the crux of the algorithm. Incoming soon.